### ISSUE TITLE
weird number of frames when writing to crd or mdcrd

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#161]
This actually happens in pytraj. 

I've fixed an issue in pytraj (https://github.com/Amber-MD/pytraj/issues/991) and got the files from jmborr.

This is the script:

```python
    def test_buffer_not_c_contiguous(self):
        # source code was lightly adapted from jmborr
        # https://github.com/Amber-MD/pytraj/issues/991
        traj = pt.load('data/issue991/short.dcd', 'data/issue991/pdb.gz',
                       mask='(!:1-256)&(@H1,@H2,@H3,@H4,@H5)')

        # Trajectory of the center of mass of the first two residues
        minitraj =  np.empty((2,traj.n_frames,3))
        minitraj[0] = pt.center_of_mass(traj,mask=':1')
        minitraj[1] = pt.center_of_mass(traj,mask=':2')
        minitraj = minitraj.transpose((1,0,2))

        # "topology" using the first two atoms
        minitop = traj.top['@1,2']

        # Save trajectory
        # make sure there is no ValueError
        # something is wrong with pdb, crd extension when loading with
        # minitop (poor topology?)
        # make issue in cpptraj too?
        for ext in ['nc', 'crd', 'mdcrd', 'dcd', 'trr']:
            fn = 'output/junk.' + ext
            pt.write_traj(filename=fn, traj=minitraj,
                          top=minitop, overwrite=True)

            ## load coord back to make sure we correctly write it
            new_traj = pt.iterload(fn, minitop)
            print('ext={0}, n_frames={1}'.format(ext, new_traj.n_frames))
```

And this is the output
```
ext=nc, n_frames=10
ext=crd, n_frames=5
ext=mdcrd, n_frames=5
ext=dcd, n_frames=10
ext=trr, n_frames=10
```

It's likely there is something weird with writing to text format. I am not sure if this is due to wrong implementation in pytraj or this is due to some dark corner in cpptraj.

datafiles are here: https://github.com/Amber-MD/pytraj/tree/master/tests/data/issue991

**Comment by hainm** [comment]
```
$ cat output/junk.crd
Cpptraj Generated trajectory
 152.212  74.175  68.871  73.910  43.151  70.485
 151.031  73.663  70.291  74.563  43.730  69.728
 151.056  73.897  69.264  73.685  43.829  70.247
 150.646  73.315  69.107  73.581  43.322  70.592
 150.548  74.317  68.613  74.218  44.216  69.740
 150.754  73.905  69.163  74.695  44.559  71.208
 152.418  74.066  69.546  74.239  44.407  69.833
 150.674  72.521  69.427  74.406  42.936  71.094
 150.091  73.269  69.179  73.178  44.087  70.793
 150.435  74.165  69.566  73.961  44.406  70.236
```

**Comment by drroe** [comment]
Should be fixed by #162

**Comment by hainm** [comment]
excellent. confirmed.
